lib/ecto/changeset/relation.ex

```diff
@@ -252,6 +251,27 @@ defmodule Ecto.Changeset.Relation do
     Changeset.change(current, changes) |> put_new_action(:update)
   end
 
+  @doc """
+  TODO
+  """
+  def on_replace(%{on_replace: :mark_as_invalid}, changeset_or_model) do
+    %{Changeset.change(changeset_or_model) | valid?: false}
```

Comment:
I'm not sure. On one hand it would be nice to signal this somehow. On the other hand this would be inconsistent. 

And what should be marked as invalid in case of `has_one`? The new changeset?